### PR TITLE
Form implementation

### PR DESCRIPTION
**Pull request by MosDevx**
#changes implemented
- add form to add post
- add form to add comments
- add functinality to add likes

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,2 +1,17 @@
 class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+
+    @comment = Comment.new(user: @current_user, post: @post, text: params[:text])
```
Comment (Deepakdanger) [inline app/controllers/comments_controller.rb:9]: - [x] [OPTIONAL] For creating any model attributes, always use strong parameters, this increases the security of data sent using forms.

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class LikesController < ApplicationController
+  def new
```
Comment (Deepakdanger) [inline app/controllers/likes_controller.rb]: - [x] The Likes should be create in `create` action, so kinldy use `def create` method.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -12,4 +12,20 @@ def show
   def custom_post
     @post = Post.find(params[:id])
   end
+
+  def new
+    puts '^^^^^^^New is ^^'
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb]: - [x] kindly remove unnecessary code.

File: config/routes.rb [hunk 4]
```diff
@@ -3,13 +3,20 @@
...
-  get '/users/:id', to: "users#show", as:'user'
 
-  get 'users/:user_id/posts', to: "posts#index" , as:'user_posts'
-  get 'users/:user_id/posts/:id', to: "posts#show" , as:'user_post'
-  # get 'users/:user_id/posts/:id', to: "posts#custom_post" , as:'custom_user_post'
+  # get '/users', to: "users#index"
+  # get '/users/:id', to: "users#show", as:'user'
+  resources :users do
+    resources :posts do
+      resources :likes, only: [:new]
```
Comment (Deepakdanger) [inline config/routes.rb]: - [x] Here kindly use `resources :likes, only: [:create]`

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -12,4 +12,20 @@ def show
...
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(user: @current_user, title: params[:title], text: params[:text])
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb:21]: - [x] [OPTIONAL] For creating any model attributes, always use strong parameters, this increases the security of data sent using forms.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @MosDevx, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @MosDevx, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/likes_controller.rb (hunk 2 above)
Comment (MosDevx) [reply app/controllers/likes_controller.rb]: Done

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (MosDevx) [reply app/controllers/posts_controller.rb]: Done

File: config/routes.rb (hunk 4 above)
Comment (MosDevx) [reply config/routes.rb]: done

**Review by Ismailco — APPROVED** [review]
### Hi @MosDevx, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**